From robert.jarzmik@free.fr  Mon Oct 20 16:06:20 2008
From: Robert Jarzmik <robert.jarzmik@free.fr>
Date: Mon, 13 Oct 2008 20:06:00 +0200
Subject: USB: Fix unneeded endpoint check in pxa27x_udc
To: greg@kroah.com
Cc: Alan Cox <alan@lxorguk.ukuu.org.uk>, <danielm77@spray.se>, linux-usb@vger.kernel.org
Message-ID: <87prm4s73r.fsf@free.fr>

From: Robert Jarzmik <robert.jarzmik@free.fr>

The request allocation code doesn't need to check if the
endpoint is not NULL, as the only caller in
include/linux/usb/gadget.h, usb_ep_alloc_request() needs the
endpoint pointer to have a correct value to trigger the
allocation code.

Signed-off-by: Robert Jarzmik <robert.jarzmik@free.fr>
Acked-by: Alan Cox <alan@lxorguk.ukuu.org.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/pxa27x_udc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/gadget/pxa27x_udc.c
+++ b/drivers/usb/gadget/pxa27x_udc.c
@@ -650,7 +650,7 @@ pxa_ep_alloc_request(struct usb_ep *_ep,
 	struct pxa27x_request *req;
 
 	req = kzalloc(sizeof *req, gfp_flags);
-	if (!req || !_ep)
+	if (!req)
 		return NULL;
 
 	INIT_LIST_HEAD(&req->queue);
